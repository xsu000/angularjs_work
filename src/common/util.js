import { env, envs } from "./global";

/*eslint no-console: ["error", { allow: ["warn", "error", "log"] }] */

export const Ramda = R;

export const Ng = angular;

export const Rx = rxjs;

export const RxOper = rxjs.operators;

export const log = function (msg) {
  if (env === envs.DEV) {
    console.log(msg);
  }
}

export const greetings = 'Hello from.. util!';

/**
 * camelCaseToDash('userId') => "user-id"
 * camelCaseToDash('waitAMoment') => "wait-a-moment"
 * camelCaseToDash('TurboPascal') => "turbo-pascal"
 */
export function camelCaseToDash(str) {
  return str.replace(/([a-zA-Z])(?=[A-Z])/g, '$1-').toLowerCase()
}

/**
 * This version will handle imperfect camel case/pascal case with multiple capital letters in a row
 * @param {string} str
 */
export function camelCaseToDashAdv(str) {
  return str
    .replace(/[^a-zA-Z0-9]+/g, '-')
    .replace(/([A-Z]+)([A-Z][a-z])/g, '$1-$2')
    .replace(/([a-z])([A-Z])/g, '$1-$2')
    .replace(/([0-9])([^0-9])/g, '$1-$2')
    .replace(/([^0-9])([0-9])/g, '$1-$2')
    .replace(/-+/g, '-')
    .toLowerCase();
}

export const generateTemplate = (camelCaseName, properties = []) => {
  const tag = camelCaseToDash(camelCaseName);
  const props = Ramda.join('', Ramda.map(o => {
    return ' ' + o.prop + '="' + o.value + '"';
  }, properties));
  return '<' + tag + props + '></' + tag + '>';
}
