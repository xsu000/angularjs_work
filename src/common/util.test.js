import { generateTemplate } from "./util";

describe('Util test', function () {

  beforeEach(() => {
  });

  // beforeEach(inject((mathService) => {
  // _mathservice = mathService;
  // }));

  test('Deep merge right empty object', function () {
    var actual = generateTemplate('chapterView', []);
    expect(actual).toMatchSnapshot();
  });

});