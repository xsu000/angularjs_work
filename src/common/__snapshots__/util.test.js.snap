// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Util test Deep merge right empty object 1`] = `"<chapter-view></chapter-view>"`;
